crates/core/tedge_agent/src/entity_manager/server.rs

```diff
@@ -125,13 +125,26 @@ impl EntityStoreServer {
         }
 
         match EntityRegistrationMessage::try_from(&message) {
-            Ok(entity) => {
-                if let Err(err) = self.entity_store.update(entity.clone()) {
+            Ok(entity) => match self.entity_store.update(entity.clone()) {
+                Ok(registered) => {
+                    for entity in registered {
+                        for (fragment_key, fragment_value) in entity.reg_message.other {
+                            let twin_channel = Channel::EntityTwinData { fragment_key };
+                            let topic = self
+                                .mqtt_schema
+                                .topic_for(&entity.reg_message.topic_id, &twin_channel);
+                            let message =
+                                MqttMessage::new(&topic, fragment_value.to_string()).with_retain();
+                            self.publish_message(message).await;
```

Comment:
Rather than a `publish_message()` method, I would add a `publish_twin_data()` method, to focus on the purpose rather than the mechanism.

Reply:
I'll add the `publish_twin_data` wrapper on top of `publish_message`. I introduced `publish_message` to be reused by all other functions in this module.